admin-dev/themes/new-theme/js/pages/module/controller.js

```diff
@@ -617,10 +617,50 @@ class AdminModuleController {
   }
 
   initAddModuleAction() {
-    const self = this;
-    const addModuleButton = $(self.importModalBtnSelector);
-    addModuleButton.attr('data-toggle', 'modal');
-    addModuleButton.attr('data-target', self.dropZoneModalSelector);
+    const body = $('body');
+    const isMaintenanceMode = window.isShopMaintenance;
+    const modal = $(`#${$(this).data('confirm_modal')}`);
+
+    body.on('click', this.importModalBtnSelector, () => {
+      if (modal.length !== 1) {
+        const maintenanceLink = document.createElement('a');
+        maintenanceLink.classList.add('btn', 'btn-primary', 'btn-lg');
+        maintenanceLink.setAttribute('href', window.moduleURLs.maintenancePage);
+        maintenanceLink.innerHTML = window.moduleTranslations.moduleModalUpdateMaintenance;
+
+        const importConfirmModal = new ConfirmModal(
+          {
+            id: 'confirm-module-import-modal',
+            confirmTitle:
+            window.moduleTranslations.singleModuleModalImportTitle,
+            closeButtonLabel: window.moduleTranslations.moduleModalCancel,
+            confirmButtonLabel: isMaintenanceMode
+              ? window.moduleTranslations.moduleModalImport
+              : window.moduleTranslations.importAnywayButtonText,
+            confirmButtonClass: isMaintenanceMode
+              ? 'btn-primary'
+              : 'btn-secondary',
+            confirmMessage: isMaintenanceMode
+              ? ''
+              : window.moduleTranslations.moduleModalImportConfirmMessage,
+            closable: true,
+            customButtons: isMaintenanceMode ? [] : [maintenanceLink],
+          },
+
+          () => $(this.dropZoneModalSelector).modal({show: true}),
+        );
+
+        importConfirmModal.show();
+      } else {
+        return (
+          self.dispatchPreEvent('import', this)
+          && self.confirmAction('import', this)
+          && self.requestToController('import', $(this))
```

Comment:
These methods don't exist in the current class, I don't know what you're supposed to do in case the modal doesn't exist but you should definitely modify this part

Reply:
Yeah it was a copy/paste from module-card.ts. Condition is useless. removed it. stil wondering what's it's use in module-card.ts